src/Sdk/WebApi/WebApi/Results/ResultsHttpClient.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using GitHub.Services.Results.Contracts;
+using GitHub.Services.OAuth;
+using System.Net.Http.Formatting;
+using Sdk.WebApi.WebApi;
+
+namespace GitHub.Services.Results.Client
+{
+    public class ResultsHttpClient : RawHttpClientBase
+    {
+        public ResultsHttpClient(Uri baseAddress, VssOAuthCredential credentials)
+            : base(baseAddress, credentials)
+        {
+            m_credentials = credentials;
+            m_resultsServiceUrl = baseAddress;
+            m_formatter = new JsonMediaTypeFormatter();
+        }
+
+        public async Task<StepSummaryUploadUrlResponse> GetStepSummaryUploadUrlAsync(string jobId, string planId, string stepId, CancellationToken cancellationToken)
+        {
+            var request = new StepSummaryUploadUrlRequest()
+            {
+                WorkflowJobRunBackendId = jobId,
+                WorkflowRunBackendId = planId,
+                StepBackendId = stepId
+            };
+
+            var stepSummaryUploadRequest = m_resultsServiceUrl+"/GetStepSummarySignedBlobURL";
+
+            HttpRequestMessage requestMessage = new HttpRequestMessage(HttpMethod.Post, stepSummaryUploadRequest);
+            // TODO: Figure out how to get the correct token here
+            requestMessage.Headers.Authorization = new AuthenticationHeaderValue("Bearer", m_credentials.InitialToken.ToString());
```

Comment:
Is this how we get the token from the VssCredentials to be able to use authentication here?